src/raftstore/store/snap.rs

```diff
@@ -618,6 +670,1025 @@ mod v1 {
     }
 }
 
+mod v2 {
+    use std::io::{self, Read, Write, ErrorKind};
+    use std::fs::{self, File, OpenOptions, Metadata};
+    use std::path::{Path, PathBuf};
+    use std::sync::{Arc, RwLock};
+    use std::str;
+    use std::time::Instant;
+    use crc::crc32::{self, Digest, Hasher32};
+    use protobuf::{Message, RepeatedField};
+    use kvproto::metapb::Region;
+    use kvproto::raft_serverpb::{SnapshotCFFile, SnapshotMeta, RaftSnapshotData};
+    use rocksdb::{EnvOptions, SstFileWriter, IngestExternalFileOptions};
+    use storage::{CfName, CF_DEFAULT, CF_LOCK, CF_WRITE};
+    use util::{HandyRwLock, rocksdb, duration_to_sec};
+    use raftstore::Result as RaftStoreResult;
+
+    use super::super::engine::{Snapshot as DbSnapshot, Iterable};
+    use super::super::keys::{enc_start_key, enc_end_key};
+    use super::super::metrics::{SNAPSHOT_CF_KV_COUNT, SNAPSHOT_CF_SIZE,
+                                SNAPSHOT_BUILD_TIME_HISTOGRAM};
+    use super::{SNAP_GEN_PREFIX, SNAP_REV_PREFIX, TMP_FILE_SUFFIX, SST_FILE_SUFFIX, Result,
+                SnapKey, Snapshot, BuildContext, ApplyOptions, check_abort, need_to_pack};
+    use super::v1::{build_plain_cf_file, apply_plain_cf_file};
+
+    pub const SNAPSHOT_VERSION: u64 = 2;
+    const META_FILE_SUFFIX: &'static str = ".meta";
+    const DIGEST_BUFFER_SIZE: usize = 10240;
+    const SNAPSHOT_CFS: &'static [CfName] = &[CF_DEFAULT, CF_LOCK, CF_WRITE];
```

Comment:
It seems there is no way to initialize a const module variable. Then `SNAPSHOT_CFS` is a must-exist, `need_to_pack` could be replaced, although `need_to_pack` is better for the possible CF modification in the future.